app/cli_options.cpp

```diff
@@ -19,6 +19,21 @@ op::options_description cli_options() {
   ad("log,l", op::value<std::string>()->default_value("info"), "Set lowest log level to show. Possible options: trace, debug, info, warning, error, fatal, none. Default: info");
   ad("first-frame", op::value<int>(), "Desired lowest start frame (inclusive).");
   ad("last-frame", op::value<int>(), "Desired highest end frame (inclusive).");
+  ad("first-stream", op::value<int>(), "Desired lowest start stream (inclusive).");
+  ad("last-stream", op::value<int>(), "Desired highest end stream (inclusive).");
+
+  // pipeline modules
+  ad("pipeline-reader", op::value<std::string>()->default_value("matlab-concurrent"), "Which reader module to use. Valid values: none, matlab, matlab-concurrent");
+  ad("pipeline-registration", op::value<std::string>()->default_value("disparity-cpu-optimized"), "Which registration module to use. Valid values: none, disparity, disparity-cpu-optimized");
+  ad("pipeline-point-cloud-filtering", op::value<std::string>()->default_value("none"), "Which filtering modules to use. Valid values: none, subsample");
+  ad("pipeline-clustering", op::value<std::string>()->default_value("mean-shift"), "Which clustering module to use. Valid values: none, mean-shift");
```

Comment:
I agree this helps us in the long run, but is this really important for us right now?

Reply:
I'm using this options really hard at the moment (I'm implementing different filters, modifying them easily is a must). I may remind you, that we are now in the improvement phase where we need to be flexible regarding algorithm choice.